src/modules/java.rs

```diff
@@ -25,12 +26,35 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         Some(java_version) => {
             let mut module = context.new_module("java");
             let config: JavaConfig = JavaConfig::try_load(module.config);
-            module.set_style(config.style);
-
             let formatted_version = format_java_version(java_version)?;
-            module.create_segment("symbol", &config.symbol);
-            module.create_segment("version", &SegmentConfig::new(&formatted_version));
 
+            let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+                formatter
+                    .map_meta(|var, _| match var {
+                        "symbol" => Some(config.symbol),
+                        _ => None,
+                    })
+                    .map_style(|variable| match variable {
+                        "style" => Some(Ok(config.style)),
+                        _ => None,
+                    })
+                    .map(|variable| match variable {
+                        // This may result in multiple calls to `get_module_version` when a user have
+                        // multiple `$version` variables defined in `format`.
+                        "version" => Some(Ok(formatted_version.clone())),
```

Comment:
Cloning isn't required here

Reply:
Without it, I get this error:
```
cannot move out of `formatted_version`, a captured variable in an `Fn` closure
  --> src\modules\java.rs:42:46
42 |                         "version" => Some(Ok(formatted_version)),
   |                                              ^^^^^^^^^^^^^^^^^ move occurs because `formatted_version` has type `std::string::String`, which does not implement the `Copy` trait
```

Reply:
Okay I changed it to a reference using `&` and it works. 👍